### DISCUSSION TITLE
Trying to update a very old cluster

### DISCUSSION THREAD

**Discussion by uyilmaz** [aws/amazon-vpc-cni-k8s#2960] [Q&A] [answered]
I have an old EKS cluster with version 1.23. I am in progress of updating it. Since it has self managed nodes, I thought I need to also update aws-node daemonset and therefore amazon-vpc-cni

The current running version is v1.7.10-eksbuild.1. I tried replacing it with v1.18.2-eksbuild.1 since it is the latest version and README says to use the latest version of vpc-cni plugin regardless of the k8s version. But when I do the update, pods fail with the following error:

`Readiness probe failed: command "/app/grpc-health-probe -addr=:50051" timed out`

I couldn't find any more information on how to update the plugin or what the requirements are for the update. Can you help?

**Comment by orsenthil** [comment, marked as answer]
@uyilmaz - the CNI upgrade is recommended one version at the time; especially from a version like v1.7.10.

**Comment by hthmkhlf** [comment]
From the official [AWS Documentation](https://docs.aws.amazon.com/eks/latest/userguide/managing-vpc-cni.html)

> To upgrade to VPC CNI v1.12.0 or later, you must upgrade to VPC CNI v1.7.0 first. We recommend that you update one minor version at a time.
